### DISCUSSION TITLE
Event driven workflows

### DISCUSSION THREAD

**Discussion by douglas-smith-techlabs** [argoproj/argo-workflows#5819] [Q&A] [answered]
Hi - sorry, really basic getting started question here.  But I can't seem to find it in the docs really.  But if I want event driven workflows managed by Argo, I need to install argo workflows -and- argo events?  When I install argo workflows, in the web interface I see events and sensors and the ability to create new events and sensors.  But also, the docs say to install events in the namespace argo-events and the workflows in the namespace argo.  If I am to install both argo events and argo workflows, should I install the argo events in the argo namespace for use with workflows?

**Comment by terrytangyuan** [comment, marked as answer]
You should install both of them in the same namespace, regardless of the specific name of the namespace. Make sure to use namespaced installation and specify --namespaced in your controllers.

**Reply by huytn1219** [reply]
You also need to install Event Bus on the Argo namespace in order for Argo Event to communicate with as well

**Reply by douglas-smith-techlabs** [reply]
Right, sure, so I need to download the argo-events manifest install yamls, and then edit them to change the namespace to argo from argo-events, and then use those to install?  Or if I use -n argo in the kubectl will this over-ride the yaml namespace value?

**Reply by huytn1219** [reply]
You can use something like this
```
---
apiVersion: argoproj.io/v1alpha1
kind: EventBus
metadata:
  name: default
  namespace: argo
spec:
  nats:
    native:
      replicas: 3
      auth: token
```
